pyphare/pyphare/pharesee/run.py

```diff
@@ -89,6 +89,42 @@ def _compute_current(patch):
                 {"name":"Jy", "data":Jy, "centering":centering["Jy"]},
                 {"name":"Jz", "data":Jz,"centering":centering["Jz"]})
 
+
+def _divB2D(Bx,By, xBx, yBy):
+    dxbx = (Bx[1:,:]-Bx[:-1,:])/(xBx[1]-xBx[0])
+    dyby = (By[:,1:]-By[:,:-1])/(yBy[1]-yBy[0])
+    return dxbx + dyby
+
+def _compute_divB(patch):
+
+    if patch.box.ndim ==1:
+        raise ValueError("divB is 0 by construction in 1D")
+
+    elif patch.box.ndim == 2:
+        By  = patch.patch_datas["By"].dataset[:]
+        Bx  = patch.patch_datas["Bx"].dataset[:]
+        xBx = patch.patch_datas["Bx"].x
+        yBy = patch.patch_datas["By"].y
+        divB = _divB2D(Bx,By, xBx, yBy)
+
+        return ({"name":"cell_custom", "data":divB, "centering":"dual"},)
+
+    else:
+        raise RuntimeError("dimension not implemented")
+
+
+
+def _get_rank(patch):
```

Comment:
using `rank` could be slightly confusing in the realm of MPI

Reply:
I don't understand your comment, it is indeed the MPI rank that's extracted.

Reply:
ok I thought it was per patch id